### ISSUE TITLE
Cosmos.HAL.Pit.PitTimer delay can't be over 2.1475 seconds (need to use ulong)

### ISSUE THREAD

**Issue by terminal-cs** [CosmosOS/Cosmos#2341]
#### Area of Cosmos - What area of Cosmos are we dealing with?
Cosmos.Hal.Pit.PitTimer

#### Expected Behaviour - What do you think that should happen?
should be able to use large numbers to represent time over 2.1475 seconds

#### Actual Behaviour - What unexpectedly happens?
can't due to int32 max limit, PitTimer only allows creation with int delays

#### Reproduction - How did you get this error to appear?
use the following code:
```cs
_ = new PIT.PITTimer(new(() => { Heap.Collect(); }), 10000000000, true);
```

#### Version - Were you using the User Kit or Dev Kit? And what User Kit version or Dev Kit commit (Cosmos, IL2CPU, X#)?
all versions

**Comment by terminal-cs** [comment]
i recomend using ulong because times below 0 are useless ad it would allow us to access very high numbers (about 3 minutes)

**Comment by ascpixi** [comment]
Until this is implemented, a workaround is possible by, for example, using a tick system:
```cs
uint tick = 0;
var timer = new PIT.PITTimer(() => {
      if(tick % 60 == 0){
           // The following will also run on the first tick of the PIT.
           Console.WriteLine("A minute has passed.");
      }
      tick++;
}, 1000000000, true); // Runs every second
Global.PIT.RegisterTimer(timer);
```

**Comment by quajak** [comment]
I think we can go ahead and change the variables from int to long in the PITTimer class.

**Comment by terminal-cs** [comment]
Why just long and not ulong?

**Comment by quajak** [comment]
You are right, we can use ulong.

**Comment by terminal-cs** [comment]
Closed: will be fixed when #2417 becomes merged.
